spring-cloud-task-core/src/main/java/org/springframework/cloud/task/listener/DefaultTaskExecutionKeyValuesProvider.java

```diff
@@ -38,17 +36,20 @@ public class DefaultTaskExecutionKeyValuesProvider implements TaskExecutionKeyVa
 
 	@Override
 	public KeyValues getLowCardinalityKeyValues(TaskExecutionObservationContext context) {
-		TaskExecution execution = context.getTaskExecution();
-		return KeyValues.of(TASK_NAME, execution.getTaskName(), TASK_STATUS, context.getStatus(),
-			TASK_PARENT_EXECUTION_ID, String.valueOf(execution.getParentExecutionId()),
-			TASK_EXIT_CODE, String.valueOf(execution.getExitCode()),
-			TASK_EXCEPTION, context.getExceptionMessage(),
-			TASK_EXECUTION_ID, String.valueOf(execution.getExecutionId()));
+		return getKeyValuesForTaskExecution(context);
 	}
 
 	@Override
 	public KeyValues getHighCardinalityKeyValues(TaskExecutionObservationContext context) {
+		return getKeyValuesForTaskExecution(context);
+	}
+
+	private KeyValues getKeyValuesForTaskExecution(TaskExecutionObservationContext context) {
 		TaskExecution execution = context.getTaskExecution();
-		return KeyValues.of(TASK_NAME, execution.getTaskName());
+		return KeyValues.of(
```

Comment:
A keyvalue should have low or high cardinality, if all of these are low, the `getHighCardinalityKeyValues` method can be revoved.